### DISCUSSION TITLE
when using carbon components sveltekit runs slow

### DISCUSSION THREAD

**Discussion by cybermedusa** [sveltejs/kit#5637] [General]
I started using carbon components in sveltekit app and I see a big difference in the time of refreshing a page.
Without carbon components, page is refreshing very quickly but with carbon, it takes approximately 20/30 seconds to refresh.
What could be causing this issue?

**Comment by dominikg** [comment]
This is likely caused by carbons index.js exporting a lot of components which are then all requested by the dev server one at a time. Check the network log.

Two possible fixes:

1) use carbon's preprocessor to rewrite your imports to deep imports: 
 
   https://github.com/carbon-design-system/carbon-preprocess-svelte
   https://github.com/carbon-design-system/carbon-components-svelte#usage-1

2) enable pre-bundling of svelte libraries 
   https://github.com/sveltejs/vite-plugin-svelte/blob/main/docs/config.md#prebundlesveltelibraries
   ```
   // svelte.config.js
   export default {
     vitePlugin: {
       experimental: {
         prebundleSvelteLibraries: true
       }
     }
   };
   ```
